src/main/java/org/prebid/server/auction/VideoRequestFactory.java

```diff
@@ -0,0 +1,504 @@
+                        .toAuctionContext(routingContext, bidRequestToPodError.getLeft(), startTime, timeoutResolver)
+                        .map(auctionContext -> Tuple2.of(auctionContext, bidRequestToPodError.getRight())));
+    }
+
+    /**
+     * Parses request body to {@link BidRequestVideo}.
+     * <p>
+     * Throws {@link InvalidRequestException} if body is empty, exceeds max request size or couldn't be deserialized.
+     */
+    private BidRequestVideo parseRequest(RoutingContext context) {
+        final Buffer body = context.getBody();
+        if (body == null) {
+            throw new InvalidRequestException("Incoming request has no body");
+        }
+
+        if (body.length() > maxRequestSize) {
+            throw new InvalidRequestException(
+                    String.format("Request size exceeded max size of %d bytes.", maxRequestSize));
+        }
+
+        try {
+            return Json.decodeValue(body, BidRequestVideo.class);
+        } catch (DecodeException e) {
+            throw new InvalidRequestException(e.getMessage());
+        }
+    }
+
+    private static Set<String> podIds(BidRequestVideo incomingBidRequest) {
+        final Podconfig podconfig = incomingBidRequest.getPodconfig();
+        if (podconfig != null) {
+            return podconfig.getPods().stream()
+                    .map(Pod::getPodId)
+                    .filter(Objects::nonNull)
+                    .map(String::valueOf)
+                    .collect(Collectors.toSet());
+        } else {
+            return Collections.emptySet();
+        }
+    }
+
+    private Future<Tuple2<BidRequest, List<PodError>>> createBidRequest(RoutingContext routingContext,
+                                                                        BidRequestVideo bidRequestVideo,
+                                                                        String storedVideoId, Set<String> podIds) {
+        return storedRequestProcessor.processVideoRequest(storedVideoId, podIds, bidRequestVideo)
+                .map(storedData -> doAndReturn(() -> validateStoredBidRequest(storedData.getStoredData()), storedData))
+                .map(this::mergeWithDefaultBidRequest)
+                .map(bidRequestToErrors -> Tuple2.of(auctionRequestFactory
+                                .fillImplicitParameters(bidRequestToErrors.getLeft(), routingContext, timeoutResolver),
+                        bidRequestToErrors.getRight()))
+                .map(requestToPodErrors ->
+                        Tuple2.of(auctionRequestFactory.validateRequest(requestToPodErrors.getLeft()),
+                                requestToPodErrors.getRight()));
+    }
+
+    private static <T> T doAndReturn(Runnable runnable, T returned) {
+        runnable.run();
+        return returned;
+    }
+
+    /**
+     * Throws {@link InvalidRequestException} in case of invalid {@link BidRequestVideo}.
+     */
+    private void validateStoredBidRequest(BidRequestVideo bidRequestVideo) {
```

Comment:
There are couple which we need to keep in mind. 

1. if I simply set to public `RequestValidator.validateSite` i need also open `ValidationException`. Also this method also validated `site.ext` which is not relevant for this workflow.
+ There are still a lot of validation in `VideoRequestFactory`

2. We can implement some kind of hierarchy of Validators.
leave general protected methods in `RequestValidator`. 
Then we Add `BidRequestValidator` which add validation of `Imp`
`VideoRequestValidator` extend `RequestValidator` and reuse methods (`validateSite / App`)
It will migrate all (or almost all) validation logic. But we need to decide, how we can handle `enforceStoredRequest` (pass from `VideoRequestFactory` to validation, or keep some validation in `VideoRequestFactory` to have some pretty method like 
``` public ValidationResult validate(T bidRequest)```